test/test-framework/go.mod

```diff
@@ -0,0 +1,40 @@
+module github.com/operator-framework/operator-sdk/test/test-framework
+
+go 1.13
+
+require (
+	github.com/go-openapi/spec v0.19.0
+	github.com/operator-framework/operator-sdk v0.12.1-0.20191112211508-82fc57de5e5b
```

Comment:
Does this become another file that needs to be checked and updated during releases? 

I'm wondering if we can incorporate `operator-sdk print-deps` in the `gen-test-framework` script to keep this up-to-date?

Reply:
It was the reason that I thought would be easier not to keep them committed.

Reply:
This [replace](https://github.com/operator-framework/operator-sdk/pull/2202/files#diff-e83dff1ad2bfd5a7235a9f23dcb62650R40) line *should* take care of that issue.

Reply:
I don't think it quite takes care of it. If I understand Go modules correctly, I think this would result in the compiled test-framework using all the resolved `require` dependencies of `github.com/operator-framework/operator-sdk v0.12.1-0.20191112211508-82fc57de5e5b` and then substituting just `operator-sdk` for the one listed in the `replace`.

Reply:
On further investigation it looks like you are correct. This shouldn't be an issue if we do
```console
$ cd test/test-framework
$ operator-sdk print-deps > go.mod
$ echo -e "\ngithub.com/operator-framework/operator-sdk => ../../" >> go.mod
$ go build ./...
$ go mod tidy
```
in `gen-test-framework`. This will avoid passing incompatible transitive dependencies to `test-framework` and make sure that pinned dependencies are up-to-date.

Reply:
I think for the propose to use the testdata as a project mock and ensure that it is updated has no need to do it. If we would like to facilitate the update of the testadata we can create a make target that will do as it is done in the check. Wdyt?